src/AutoRest.CSharp/Common/Output/Models/RestClient.cs

```diff
@@ -29,22 +29,21 @@ public RestClient(OperationGroup operationGroup, BuildContext context, string? c
             OperationGroup = operationGroup;
             Builder = new RestClientBuilder(operationGroup, context);
 
-            _requestMethods = new CachedDictionary<ServiceRequest, RestClientMethod> (EnsureNormalMethods);
-            _nextPageMethods = new CachedDictionary<ServiceRequest, RestClientMethod> (EnsureGetNextPageMethods);
+            _requestMethods = new CachedDictionary<ServiceRequest, RestClientMethod>(EnsureNormalMethods);
+            _nextPageMethods = new CachedDictionary<ServiceRequest, RestClientMethod>(EnsureGetNextPageMethods);
 
-            Parameters = Builder.GetOrderedParameters ();
+            Parameters = Builder.GetOrderedParameters();
 
             ClientPrefix = ClientBuilder.GetClientPrefix(clientName ?? operationGroup.Language.Default.Name, context);
-            RestClientSuffix = "Rest" + ClientBuilder.GetClientSuffix(context);
+            RestClientSuffix = ClientBuilder.GetRestClientSuffix(context);
             DefaultName = ClientPrefix + RestClientSuffix;
-            Description = "";
         }
 
         protected RestClientBuilder Builder;
         internal OperationGroup OperationGroup { get; }
         protected string RestClientSuffix { get; }
-        public Parameter[] Parameters { get; }
-        public string Description { get; }
+        public virtual Parameter[] Parameters { get; }
```

Comment:
I'm not sure I understand why it has to be virtual?

Reply:
I don't think they need to be either - possibly this is just left over from some exploratory stuff I was doing.  I will remove.  Thanks for pointing this out!

Reply:
Actually - I take it back after I tried to undo it.  The issue is that `LowLevelRestClient` (a subclass of this) needs to override the `Parameters` to remove the api-version parameter (since it instead is handled as part of the client options).  You can see that `LowLevelRestClient.cs` provides implementations for both this and `Description`,

Reply:
FYI - The changes I am currently working on for Pageable should allow me to remove this.

Reply:
This did get cleaned up with my Pageable changes.  Going to leave this unresolved for now but it will get fixed up with my next PR.